ports/renesas-ra/Makefile

```diff
@@ -429,13 +470,27 @@ $(PY_BUILD)/formatfloat.o: COPT += -Os
 $(PY_BUILD)/parsenum.o: COPT += -Os
 $(PY_BUILD)/mpprint.o: COPT += -Os
 
-all: $(TOP)/lib/fsp/README.md $(BUILD)/firmware.hex
+all: $(TOP)/lib/fsp/README.md CHECK_FSP_VER $(BUILD)/firmware.hex
 
 # For convenience, automatically fetch required submodules if they don't exist
 $(TOP)/lib/fsp/README.md:
 	$(ECHO) "fsp submodule not found, fetching it now..."
 	(cd $(TOP) && git submodule update --init lib/fsp)
 
+CURRENT_FSP_HASH = $(patsubst +%,%,$(shell git -C $(TOP) submodule status lib/fsp))
+CHECK_FSP_VER:
+ifneq ($(CURRENT_FSP_HASH),$(EXPECTED_FSP_HASH))
+ifeq ($(BOARD),VK_RA6M5)
```

Comment:
```suggestion
ifeq ($(CMSIS_MCU),RA6M5)
```

Reply:
OK that makes sense, It will be fixed.

Reply:
On second thought, since the boards are tightly coupled to the FSP version used in the e2studio, I have a better suggestion here, maybe just set `EXPECTED_FSP_HASH` in `mpconfigboard.mk` ? This way every board can override the default *if needed*. In the Makefile you can just set the default if it's not defined:

```Makefile
EXPECTED_FSP_HASH ?= $(patsubst +%,%,$(shell git -C ../.. submodule status --cached lib/fsp))
```

Reply:
Great Idea I will be happy to fix it that way.

Reply:
Just noticed that switching between two boards that require two different FSPs *with* parallel builds (`make -j<n>`) seems to fail on the first run (only the first run), so it seems that some objects start building before FSP is switched. Making the FSP check an order dependency of `$(OBJ)` seems to fix it:

```Makefile
$(OBJ): | CHECK_FSP_VER $(GEN_PINS_HDR)
```

Reply:
What a great idea!  It was a headache for me. Thank you @iabdalkader and @mbedNoobNinja .